Update federal.mdx (#262)
launchdarkly/LaunchDarkly-Docs: src/content/topics/home/advanced/federal.mdx

<Callout intent="info">
<CalloutTitle>Federal instance URLs end in ".us"</CalloutTitle>
<CalloutDescription>
  

> The federal instance does not use the common ".com" URL path. Instead, it uses the US government's path: ".us." When you navigate to LaunchDarkly Federal application, use the federal instance API, or initialize your SDK to the federal instance, you must replace the generally available ".com" URLs with URLs that end in ".us." Initialization of SDKs to the federal instance is performed in the initialization "options" settings for the SDK. Configuration options can be found in the generated API reference for each SDK. For example, the initialization options for the Node SDK are here. Depending on the SDK, there will be two or three service endpoints required to be configured in the initialization options: base/polling endpoint, streaming endpoint, and an events endpoint. Note the option key value pair names may differ between SDKs as we respect each language's URI/URL naming conventions and supported services. So, it is critical to reference the documentation for each specific SDK. Below are two sample codeblocks for reference to highlight differences.

Reviewer: ```suggestion
The federal instance does not use the common ".com" URL path. Instead, it uses the US government's path: ".us." When you navigate to LaunchDarkly Federal application, use the federal instance API, or initialize your SDK to the federal instance, you must replace the generally available ".com" URLs with URLs that end in ".us." 

Initialization of SDKs to the federal instance is performed in the initialization "options" settings for the SDK. Configuration options can be found in the generated API reference for each SDK. 

For example, the initialization options for the Node SDK are here. Depending on the SDK, there may be multiple service endpoints required to be configured in the initialization options. These could include the base/polling endpoint, streaming endpoint, and an events endpoint. 

Note the option key value pair names may differ between SDKs as we respect each language's URI/URL naming conventions and supported services. So, it is critical to reference the documentation for each specific SDK. 

Sample code is provided below for reference to highlight differences.
```

Reviewer 2: ```suggestion
The federal instance does not use the common ".com" URL path. Instead, it uses the US government's path: ".us." When you navigate to LaunchDarkly Federal application, use the federal instance API, or initialize your SDK to the federal instance, you must replace the generally available ".com" URLs with URLs that end in ".us." 

You can initialize an SDK for the federal instance using the initialization "options" settings for the SDK. You can also find configuration options in the generated API reference for each SDK. 

Depending on the SDK, it may require you to configure multiple service endpoints in the initialization options. These could include the base and polling endpoint, streaming endpoint, and an events endpoint. For example, you can read the [initialization options for the Node SDK](TKTK should this link to our Node config page?). 

The option key value pair names may differ between SDKs, as LaunchDarkly respects each language's URI and URL naming conventions and supported services. For this reason, it is critical to reference the documentation for each individual SDK. 
```
Changes to active voice.

>   React example:

Reviewer 2: ```suggestion
Here is an example for the React SDK:
  
  <CodeTabs
defaultValue="json"
values={[
  { label: 'json', value: 'json', },
]
}>
<CodeTabItem value="json">

```json
```

  async () => {
  // Set clientSideID to your own Client-side ID. You can find this in
  // your LaunchDarkly portal under Account settings / Projects
  const LDProvider = await asyncWithLDProvider({
     clientSideID: 'your-client-side-ID-here',
     user: {
       "key":"samplekey", "name":"First Last", "email":"example@yourdomain.com"
     },
    options: {
       baseUrl:"https://sdk.launchdarkly.us",
       streamUrl:"https://clientstream.launchdarkly.us",
       eventsUrl:"https://events.launchdarkly.us"
    },
    deferInitialization: false
});
 
  Java example (server-side SDKs normally require streaming and events settings, not polling):
 
LDConfig config = new LDConfig.Builder()
  .dataSource(Components.streamingDataSource()
    .baseURI(URI.create("https://stream.launchdarkly.us"))
  .events(Components.sendEvents()
    .baseURI(URI.create("https://events.launchdarkly.us"))

>   .build();

Reviewer: These should be `code` formatted (with the exception of line 51 of course), but I can't seem to insert that markdown as a suggestion

Reviewer: We resolved this, with the exception of the CodeTabs and CodeTabItems

>  

Reviewer 2: The three backticks tend to trip up Github comments, because the comments use the same syntax. This code sample needs the below closing tags plus some closing backtics, but we can add those manually later instead of via a comment since they don't display here.
```suggestion
</CodeTabItem>
</CodeTabs>
```

Reviewer: We couldn't quite figure out how to close these tags properly, so could use your help on it 😄

> Additional language syntax examples for setting configuration options can be found in the documentation for LaunchDarkly relay proxy here (sdk/features/relay-proxy-configuration/proxy-mode), just change the relay proxy URL to the respective federal instance URLs instead. A list of available service endpoints for federal are below.

Reviewer: ```suggestion
Additional language syntax examples for setting configuration options can be found in the documentation for LaunchDarkly relay proxy [here](sdk/features/relay-proxy-configuration/proxy-mode), just change the relay proxy URL to the respective federal instance URLs instead. A list of available service endpoints for federal are below.
```

* [The debugger](/home/flags/debugger)
* [Support for the Roku SDK](/sdk/client-side/roku)

> * Flag insights (home/flags/insights)

Reviewer: ```suggestion
* [Flag insights](home/flags/insights)
```
